src/libcore/num/mod.rs

```diff
@@ -2504,76 +2504,172 @@ impl fmt::Display for TryFromIntError {
     }
 }
 
-macro_rules! same_sign_try_from_int_impl {
-    ($storage:ty, $target:ty, $($source:ty),*) => {$(
+// no possible bounds violation
+macro_rules! try_from_unbounded {
+    ($source:ty, $($target:ty),*) => {$(
         #[unstable(feature = "try_from", issue = "33417")]
         impl TryFrom<$source> for $target {
             type Error = TryFromIntError;
```

Comment:
I think the infallible ones are moving to a blanket impl, per https://github.com/rust-lang/rust/issues/33417#issuecomment-306231905